openshift-provision/openshift-cluster-create.yaml

```diff
@@ -0,0 +1,119 @@
+      description: Platform name where the openshift cluster needs to be deployed such as AWS, GCP etc
+      default: ""
+    - name: PROJECT_ID
+      type: string
+      default: ""
+      description: Project Id required in case of gcp
+    - name: CLUSTER_NAME
+      type: string
+      description: Cluster name
+    - name: BASE_DOMAIN
+      type: string
+      description: Base domain of your cluster install
+    - name: REPLICAS
+      description: How many replicas for the master and worker node
+      type: string
+    - name: REGION
+      type: string
+      description: AWS Region where to install
+  steps:
+    - name: openshift-install-config
+      image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+      workingDir: $(workspaces.install-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
+        echo "1 $(ls -a)"
+        credsPath=""
+        projectId=""
+        if [ "$(params.PLATFORM)" == "gcp" ]; then
+          credsPath+=".gcp"
+          mkdir .gcp
+          projectId+="ProjectID: $(params.PROJECT_ID)"
+        else
+          credsPath+=".aws"
+          mkdir .aws
+        fi
+
+        cp $(workspaces.secrets.path)/* $credsPath
+
+        echo "2 $(ls -a $credsPath/)"
+
+        cat $credsPath/credentials
+
+        [[ -f install-config.yaml ]] && \
+        echo 'using existing install-config.yaml' && \
+        cat install-config.yaml && exit 0
+
+        cat << EOF > install-config.yaml
+        apiVersion: v1
+        baseDomain: $(params.BASE_DOMAIN)
+        compute:
+        - hyperthreading: Enabled
+          name: worker
+          platform: {}
+          replicas: $(params.REPLICAS)
+        controlPlane:
+          hyperthreading: Enabled
+          name: master
+          platform: {}
+          replicas: $(params.REPLICAS)
+        metadata:
+          creationTimestamp: null
+          name: $(params.CLUSTER_NAME)
+        networking:
+          clusterNetwork:
+          - cidr: 10.128.0.0/14
+            hostPrefix: 23
+          machineCIDR: 10.0.0.0/16
+          networkType: OpenShiftSDN
+          serviceNetwork:
+          - 172.30.0.0/16
+        platform:
+          $(params.PLATFORM):
+            region: $(params.REGION)
+            ###ProjectID
+        pullSecret: '$PULL_SECRET'
+        sshKey: '$PUBLIC_SSH_KEY'
+        EOF
+
+        sed -i "s|###ProjectID|$projectId|" ./install-config.yaml
+
+        [[ -f install-config.yaml ]] && cat install-config.yaml
+        [[ -f install-config.yaml ]] || echo skipping settings
+      env:
+        - name: PULL_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: openshift-install
+              key: pull-secret
+        - name: PUBLIC_SSH_KEY
+          valueFrom:
+            secretKeyRef:
+              name: openshift-install
+              key: public-ssh-key
+    - name: openshift-cluster-create
+      workingDir: $(workspaces.install-dir.path)
+      image: quay.io/openshift-pipeline/ci:bootstrap
```

Comment:
this image may break (as I don't provide SLA for it as of yet) and fat for other reasons... is there any others image with openshift-install you can use?

Reply:
Okay...will try searching for some other image

Reply:
@chmouel I tried using the image from https://quay.io/repository/openshift/origin-upi-installer?tab=tags
but it's not working with openshift

Reply:
I got that fixed :sunglasses: